### PR TITLE
Timed Exercise 10/25 - Open Mic - Alex Barnes

### PR DESCRIPTION
**Pull request by abarnes26**

### CODE REVIEW

File: lib/user.rb [hunk 1]
```diff
@@ -1,7 +1,33 @@
...
+    end
+  end
+
+  def tell(user, joke)
+    user.jokes << joke
+  end
+
+  def perform_routine_for(user)
+    @jokes.map do |joke|
+      tell(user, joke)
```
Comment (VictoriaVasys) [inline lib/user.rb]: Great use of your already-functional code here!

File: lib/user.rb [hunk 2]
```diff
@@ -1,7 +1,33 @@
...
+    end
+  end
+
+  def learn_routine(file = "./jokes.csv")
+    jokes = CSV.open file, headers: true
+    jokes.map do |joke|
+      unless @jokes.include?(joke)
+      @jokes << joke
+      end
```
Comment (VictoriaVasys) [inline lib/user.rb]: Take a look at the rest of code already available to you; looks like you're repeating yourself here. Also pay careful attention to indentation.

File: test/user_test.rb [hunk 3]
```diff
@@ -21,4 +22,71 @@ def test_it_can_have_another_name
...
+    joke_1 = Joke.new({id: 1,
+                 question: "Why did the strawberry cross the road?",
+                   answer: "Because his mother was in a jam."})
+    casey.learn(joke_1)
+    casey.learn(joke_1)
+    casey.learn_routine
+    casey.learn_routine
+
+    assert_equal 101, casey.jokes.count
+  end
```
Comment (VictoriaVasys) [inline test/user_test.rb]: Great edge-case testing here!

File: test/user_test.rb [hunk 4]
```diff
@@ -21,4 +22,71 @@ def test_it_can_have_another_name
...
+    assert_equal 1, sal.jokes.count
+  end
+
+  def test_user_can_tell_jokes_to_other_user
+   sal = User.new("Sal")
+   ali = User.new("Ali")
+   joke = Joke.new({id: 1,
+              question: "Why did the strawberry cross the road?",
+                answer: "Because his mother was in a jam."})
+   sal.learn(joke)
```
Comment (VictoriaVasys) [inline test/user_test.rb]: Is this method call necessary for the functionality of #tell?

File: test/user_test.rb [hunk 5]
```diff
@@ -21,4 +22,71 @@ def test_it_can_have_another_name
...
+                   answer: "Because his mother was in a jam."})
+    joke_2 = Joke.new({id: 2,
+                 question: "How do you keep a lion from charging?",
+                   answer: "Take away its credit cards."})
+    iliana.learn(joke_1)
+    iliana.learn(joke_2)
+    iliana.perform_routine_for(josh)
+
+    assert_instance_of Joke, josh.jokes.first
+    assert_equal 2, josh.jokes.count
```
Comment (VictoriaVasys) [inline test/user_test.rb]: You can also assert more specific things about the joke you expect to be included, e.g. `assert_equal "Take away its credit cards.", josh.jokes.last.answer`

**Comment by abarnes26** [comment]
Thanks for the feedback Victoria!
